### PR TITLE
build fails if final comment exceeds max length

### PR DESCRIPTION
**Pull request by teacurran**
The github API rejects comments that are more than 262144 characters.
With a large commit this will cause the build analysis fail. 

The stack trace logged is:
```
Caused by: java.io.IOException: {"message":"Validation Failed","errors":[{"resource":"IssueComment","code":"custom","field":"body","message":"body is too long (maximum is 262144 characters)"}],"documentation_url":"https://developer.github.com/v3/issues/comments/#create-a-comment"}
	at org.kohsuke.github.Requester.handleApiError(Requester.java:502)
	at org.kohsuke.github.Requester._to(Requester.java:245)
	at org.kohsuke.github.Requester.to(Requester.java:191)
	at org.kohsuke.github.GHIssue.comment(GHIssue.java:145)
	at org.sonar.plugins.github.PullRequestFacade.addGlobalComment(PullRequestFacade.java:242)
	... 55 more
```

This commit truncates the commit comment and ends it with the string "... maximum comment length 262144 reached." which isn't ideal but allows the build to succeed.

### CODE REVIEW

**Comment by henryju** [comment]
Thanks for reporting this issue. I'm a bit concerned about hardcoding the limit of 262144 in the plugin. It doesn't seem to be documented so I guess it may change without notice.
Anyway I have created a ticket:
https://jira.sonarsource.com/browse/SONARGITUB-15

**Comment by teacurran** [comment]
good point. do you think it's better to try to catch the IOException and fail the build? The reason I didn't do that is because there is no guaranteed way to recover.

you're right about the limit not being documented, that's kind of annoying. We could set a much lower limit to protect against changes and break the message into multiple comments but that isn't ideal either. This is a bit of an edge case because the limit is pretty high. My team has hundreds of pull requests that worked fine and this only came up when there was a major refactor where hundreds of classes were moved from one package to another.

**Comment by henryju** [comment]
I feel that having more than (10 ? 50 ?) issues in this comment is useless anyway. So we could indeed truncate to a certain number of issues. I'll discuss this point internally and collect some feedback.

**Comment by teacurran** [comment]
yeah, I agree that any more than 20 or so is useless.  The particular commit that broke our build was moving about 100 service classes of a legacy application to a new package.  It reported 286 major, 122 minor, 184 info issues, but none of them were on any lines in the commit so it just sorts of lists them all. This much info isn't very useful to anyone doing a code review.

**Comment by henryju** [comment]
We have decided to take the following option: keep the first 10 issues in this global comment. If more than 10 issues should have been displayed in the global comment, we could display [...] Would you like to update your pull request accordingly?

**Comment by teacurran** [comment]
updated to only show 10 issues. if there are more than ten it adds another bullet that says "... x more"

**Comment by henryju** [comment]
Cool, thanks! Would you mind merging all 3 commits in a single one for better Git timeline? Also a JUnit test would be perfect.

**Comment by teacurran** [comment]
No problem. let me know if this works for you.

File: src/test/java/org/sonar/plugins/github/GlobalReportTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import org.junit.Test;
+import org.sonar.api.issue.internal.DefaultIssue;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rule.Severity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class GlobalReportTest {
+
+	private static final String GITHUB_URL = "https://github.com/SonarCommunity/sonar-github";
```
Comment (henryju) [inline src/test/java/org/sonar/plugins/github/GlobalReportTest.java]: Indentation is not correct regarding our coding style. Could you please use configuration available in this repo: https://github.com/SonarSource/sonar-developer-toolset

**Comment by henryju** [comment]
Apart formatting issue this looks good to me. If you could update one more time with good formatting that would be perfect.

**Comment by teacurran** [comment]
I assume by good formatting you mean spaces on GlobalReportTest? If so, that is fixed. My IDE auto matches styles but since that was a new file it reverted to my default.  If there is any other formatting you need, let me know, or if you have a style guide that would be helpful.

**Comment by henryju** [comment]
LGTM

**Pull request merged**